sssd_test_framework/roles/ipa.py

```diff
@@ -641,6 +646,122 @@ def passkey_remove(self, passkey_mapping: str) -> IPAUser:
         self._exec("remove-passkey", [passkey_mapping])
         return self
 
+    def idoverrideuser(self, name:str) -> IPAUserIDOverride:
+        user_obj = IPAUser(self.role, name)
+        print("IN the idoverrideuser")
+        return IPAUserIDOverride(self.role, user_obj)
+
+        #return IPAUserIDOverride(self, name)
+
+
+class IPAUserIDOverride(IPAUser):
+    """
+    IPA User ID override management.
+    """
+
+    def __init__(self, role: IPA, user: IPAUser) -> None:
+        """
+        :param role: IPA role object.
+        :type role: IPA
+        :param name: Username.
+        :type name: strc
+        """
+        super().__init__(role, user.name)
+
+        self.name = user.name
+
+        """Anchor name"""
+
+
+    def add_id_override(
+        self,
+        view_name: str,
+        *,
+        description: str | None = None,
+        login: str | None = None,
+        uid: int | None = None,
+        gid: int | None = None,
+        gecos: str | None = None,
+        home_directory: str | None = None,
+        shell: str | None = None,
+        ssh_public_key: str | None = None,
+        certificate: str | None = None,
+    ) -> IPAUserIDOverride:
+        """
+        Add an ID override for an IPA user.
+
+        :param uid: Override for user ID, defaults to None
+        :type uid: int | None, optional
+        :param gid: Override for group ID, defaults to None
+        :type gid: int | None, optional
+        :param gecos: Override for GECOS, defaults to None
+        :type gecos: str | None, optional
+        :param home: Override for home directory, defaults to None
+        :type home: str | None, optional
+        :param shell: Override for login shell, defaults to None
+        :type shell: str | None, optional
+        :return: Self.
+        :rtype: IPAUserIDOverride
+        """
+
+
+        self.role.host.conn.run(
+            f"ipa idoverrideuser-add {view_name} {self.name}"
+            + (f' --desc="{description}"' if description else "")
+            + (f' --login="{login}"' if login else "")
+            + (f' --uid="{uid}"' if uid else "")
+            + (f' --gidnumber="{gid}"' if gid else "")
+            + (f' --gecos="{gecos}"' if gecos else "")
+            + (f' --homedir="{home_directory}"' if home_directory else "")
+            + (f' --shell="{shell}"' if shell else "")
+            + (f' --sshpubkey="{ssh_public_key}"' if ssh_public_key else "")
+            + (f' --certificate="{certificate}"' if certificate else "")
+        )
+        return self
+
+    def modify_id_override(
```

Comment:
Would rename it to ```modify``` and change the execution to ```host.conn.run or host.conn.exec``` so you don't have to sync the name of the function.

Reply:
So, after some testing the same function name is overloading the IPAUser.add() function. I know this was my suggestion, and I'm mistaken. We can rename the function to __add() to avoid the overloading.